extensions/wikia/Wall/maintenance/CleanupCommentsIndex.php

```diff
@@ -10,31 +10,66 @@ public function __construct() {
 	}
 
 	public function execute() {
+
+		$dbMaster = wfGetDB( DB_MASTER );
+		$dbSlave = wfGetDB( DB_SLAVE );
+
+		$this->updateDeletedFlag( $dbMaster, $dbSlave );
+		$this->removeBrokenRows( $dbMaster, $dbSlave );
+	}
+
+	public function updateDeletedFlag( DatabaseMysqli $dbMaster, DatabaseMysqli $dbSlave ) {
 		global $wgDBname;
 
-		$db = wfGetDB( DB_MASTER );
+		$ids = $dbSlave->selectFieldValues(
+			[ 'archive', 'comments_index' ],
+			'comment_id',
+			[ 'archived=0', 'deleted=0', 'removed=0', 'comment_id=ar_page_id', 'ar_namespace in (500,501,1200,1201,2000,2001)', 'comment_id not in (SELECT page_id from page)' ],
```

Comment:
I think it's enough to check `deleted` flag here. The other two aren't really delete options (thread closing/removal).